client/components/payment-method-details/index.js

```diff
@@ -12,14 +13,19 @@ const PaymentMethodDetails = ( props ) => {
 	if ( ! paymentMethod ) {
 		return <span>&ndash;</span>;
 	}
+	const paymentMethodAbbr = paymentMethod.last4
+		? paymentMethod.last4
+		: paymentMethod.bank_code;
 	const brand = paymentMethod.brand ? paymentMethod.brand : payment.type;
 	return (
 		<span className="payment-method-details">
 			<span
 				className={ `payment-method__brand payment-method__brand--${ brand }` }
 			/>
-			&nbsp;••••&nbsp;
-			{ paymentMethod.last4 }
+			{ paymentMethod.last4 && (
+				<Fragment>&nbsp;&bull;&bull;&bull;&bull;&nbsp;</Fragment>
+			) }
+			{ paymentMethodAbbr }
```

Comment:
(minor) could be shortened without duplicating the check on `paymentMethod.last4` on L16 by
```suggestion
			{ paymentMethod.last4 ? (
				<Fragment>&nbsp;&bull;&bull;&bull;&bull;&nbsp; { paymentMethod.last4 }</Fragment>
			) : ( paymentMethod.bank_code ) }
```